### DISCUSSION TITLE
Ender3 V2/S1 Professional Firmware 20220419 - APRIL2022 release

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Ender3V2S1#127] [Announcements]
Universal APRIL 2022 Edition
==================

This firmware is based on Marlin, has been optimized and has enhancements for the Creality Ender 3 V2/S1 printer. This is an **universal firmware** that can be used in both RET6 and RCT6 SoC chips that have 256k or 512k Flash memory. 

## Versions
- **Ender 3v2 Manual Mesh**: if you have the Ender 3v2 near to stock then use this version.
- **Ender 3v2 BLTouch**: use this version if you installed a CR/3D/BLTouch level sensor in the Ender 3v2.
- **Ender 3S1**: use this version if you have an Ender 3S1 printer

## UBL versions
Binaries with UBL in their filename have support for [Marlin Unified Bed Leveling](https://marlinfw.org/docs/features/unified_bed_leveling.html), this system is similar to the bilinear leveling but with a few extra features to provide users with more control. There are multiple guides on the web about UBL, we recommend reading about it before using this new leveling system.

* https://marlinfw.org/docs/features/unified_bed_leveling.html
* https://all3dp.com/2/marlin-ubl-unified-bed-levelin-simply-explained/
* https://www.3dmakerengineering.com/blogs/3d-printing/unified-bed-leveling-marlin
* https://hackaday.com/2022/01/14/3d-printering-getting-started-with-universal-bed-leveling/
  
<img align="left" src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main.jpg" width="270" height="480" />
<img  src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main-iconset.jpg" width="270" height="480" />

The binary files of this release are up to date and you can download them from the assets at the bottom of this page. To know if you have the last release go to the info page in the display of your printer and compare the release build date with the name of the binary file. The filename follows the convention:  
  
Ender3v2-422-ManualMesh-**YYYYMMDD**.bin
  
Where YYYYMMDD indicates the date of the build. 422 refers to the mainboard version **4.2.2**, and 427 is for the **4.2.7** mainboard version, if you got a **4.3.1** board use the 422 version. 

## Special versions
There are special compiles (as V6/Volcano) here: https://github.com/mriscoc/Marlin_Configurations/releases

<img src="https://cdn.shopify.com/s/files/1/0046/3781/8929/files/1-11.jpg" width="320" height="320" />

## Community links
* [Telegram](https://t.me/ender3v2s1firmware)
* [Reddit](https://www.reddit.com/r/Ender3v2Firmware) 
* [Ender 3V2 Facebook](https://www.facebook.com/groups/ender3v2firmware)
* [Ender 3S1 Facebook](https://www.facebook.com/groups/ender3s1printer)

## Donations
Thank you for your support, I receive donations through [Patreon](https://www.patreon.com/mriscoc) and [Paypal](https://www.paypal.com/paypalme/mriscoc)   

[<img src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif">](https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD)

# Firmware features

* Universal RCT6/RET6 Ender 3 V2/S1 Support
* DWIN / DACAI screens support
* Improved support for Manual Mesh and CR/3D/BLTouch
* User configurable power lost recovery enabler
* Manual and assisted bed tramming
* Tramming Wizard for BL/3D/CRTouch versions
* Long filename support with scroll
* Baby Z-Stepping
* Firmware Retract
* Advanced pause feature
* Configurable park-head position (default x=240 y=220)
* Filament change support
* Filament management (load/unload/park head) in prepare menu
* Print Flow on Motion and Tune menu 
* Home axis after motor disabling
* Fastest Z feed rate
* Live move for all motors but extruder
* Support for S-curve Acceleration
* Reboot printer item on control menu
* Increase extruder feed rate limit to 150 for BMG users
* Allows to invert direction of the extruder on Motion menu
* Support for BTT Smart filament sensor
* Selectable runout filament sensor active state (LOW/HIGH/MOTION)
* Editable runout distance
* Support for µSD extensions
* PID popup with temperature plot
* Lock screen
* Editable physical dimensions
* Mesh viewer
* Custom pre-heat
* Enable/disable beeps sound
* New print g-code preview (use [this ](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview)guide)
* New print done screen
* Two decimal positions for axis coordinates value
* New Z-Offset wizard
* New PID settings menus
* PID moves the head to the center for more accuracy
* Leveling bed temperature linked to bed PID temperature
* Editable extrude minimum temperature
* Brightness settings on tune menu
* Turn off the backlight of the lcd in the tune menu
* New status line system, allows the scroll of large messages
* Home submenu with individual axis homing
* Enabled user setting for 115200 bauds (Creality Box compatibility)
* Head parking before recovery after power failure
* Offset icon changes to show leveling status
* Live end-stop diagnostic page
* Move bed to front after cancel a print.
* Editable firmware retracts settings for Tune and filament settings menu
* Print Statistics page accessible from the Advanced Settings menu

![latest release](https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/New-beta.gif)

### Manual mesh
* Assisted manual bed tramming
* Assisted manual mesh leveling menu
* Selectable mesh points from 2x2 to 9x9
* Mesh viewer
* Persistent Z-offset
* Move Z to 0 and disable motors option

### CR/3D/BLTouch
* Safe Probe Z-Homing
* Bed auto level
* High speed probe
* Retry bed level on error
* Mesh viewer
* Assisted tramming
* Selectable mesh points from 2x2 to 9x9
* Allows to the user to cancel an auto leveling process 
* Increase Max Probe Offset and Probing Margin to 60 mm
* M48 Probe Test to Probe Settings menu
* Probe Stow/Deploy in Probe Settings menu
* Option to disable HS mode for better compatibility with CRTouch
* Option to change the Z Probe feed rate
* Configurable Bed auto leveling temperature

>Use the 2x2 mesh to level you bed!

<a href="https://youtu.be/qzYJC2AHbW8"><img src="https://img.youtube.com/vi/qzYJC2AHbW8/maxresdefault.jpg" width="640" height="400" /></a>  
  
### G-Codes enhanced support
* Filament Load Unload G-codes (M701/M702)
* Support for M600 filament change
* Support for G-code M117 messages
* Enable M575 change baud rate by G-code
* Extended M75 with filename support, use M75 {{ event.name }} in g-code script
* Enable ARC and 'P' parameter to specify complete circles (G2/G3)
* G35 support
* M155 Geometry report
* Native support of marlin brightness and M256 g-code

### Support for special nonstandard custom C G-Codes
* Implemented C11 En Rn Gn Bn gcode for colorize UI elements
* C108 to cancel screens waiting for user simple confirmation
* C29 N# to set amount of grid points of the mesh leveling
* C100 X### Y### to setup minimum limits
* C101 X### Y### z### to setup maximum limits
* C102 X### Y### to setup bed size
* C125 X### Y### z### to setup the park position
* C412 M# set run-out sensor active state (M0:LOW, M1:HIGH, M2:MOTION)
* C510 to lock/unlock the screen (C510 U1 to unlock)
* C562 E# to invert the Extruder (E1 to invert)
* C851 M### F#### to set probing margin, Z feed rate

<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/themes/Select_Colors.jpg" width="270" height="480" /> 

### Advanced settings menu
* Home XYZ offsets menu
* Probe offsets menu (BLTouch version)
* Customizable park-head position
* Customizable bed size and z heigh up to 999 x 999 x 999
* Launch PID for nozzle and bed with graphical feedback
* Filament run-out sensor with active state setup (LOW/HIGH)
* Filament load and unload lengths menu items in run-out menu
* Power-loss recovery enable/disable option
* LCD Brightness control
* Select Color Sub Menu
* Beep/Sound enable/disable option
* Selectable mesh point (BLTouch version)
* Mesh viewer
* Lock screen

### Special Octoprint / Host print support
* High speed (250000 baud) serial communication
* In screen control of host printing (M73, M75-M77) 
* Display messages on status line (M117)
* Color change of UI elements for visual status report (C11)
* Binary file transfer for remote firmware upgrade
* List binary (.bin) files (M20) for flash firmware support with BFT
* Arc Welder Octoprint plugin support
* EEPROM Editor Octoprint plugin support
* Host action commands for filament runout
* Emergency Parser active
* Advanced OK

**Octoprint bed visualizer support**
<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/Bed-visualizer.png" width="350" height="225" />

### Display enhancements
* Extended DWIN library
* Enhanced status messages
* Improved status bar with temperature, flow, fan, z-offset and xyz positions
* PID auto tune progress on display with plot
* Leveling progress on display
* Boot screen for M997
* Transparent icons for better color theme adaptability
* New print g-code preview
* New print done screen

**A lot of additions and fixes from Bugfix development branch**

## More info on the wiki pages ##
- [**How to install the firmware**](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)

- [**Octoprint settings**](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)

- [**How to make the Manual Mesh**](https://github.com/mriscoc/Ender3V2S1/wiki/Manual-Mesh)

- [**How to install a 3D/BLTouch**](https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch)

- [**How to apply a color theme**](https://github.com/mriscoc/Ender3V2S1/wiki/Color-Themes)

## What's new in this release ##
* New toolbar with 11 configurable shortcuts actions.
* New Advanced / Mesh Leveling menu
* Versions with **extended UBL support** (tilt, smart fill, multiple slots, etc.)
* Mesh editor for fine tuning
* Automatic maximize leveling area
* Automatic center leveling area
* Menu item for setting the default mesh fade height
* Consolidates firmware retract options in a submenu
* Support in source code of control menus for LED lights as the Neopixel
* Improves g-code preview stability
* Experimental support for Ender3S1-STM32F401RCT6  

https://user-images.githubusercontent.com/2745567/143180943-f69f5967-34c1-421f-86a1-466ecccab97d.mp4

## Precompiled firmware

<hr /><em>This discussion was created from the release <a href='https://github.com/mriscoc/Ender3V2S1/releases/tag/20220419'>Ender3 V2/S1 Professional Firmware 20220419 - APRIL2022 release</a>.</em>

**Comment by leberschnitzel** [comment]
just as fyi: Still doesn't work with S1 Pro

**Reply by mriscoc** [reply]
Sorry but maybe this firmware will never work with the S1 pro because of the touch screen.

**Reply by leberschnitzel** [reply]
oh yeah, not meant to push for that, but I'll keep trying to find a way ^^

**Comment by Dooti-2** [comment]
Just flashed -- perfect.  Love the sub-menu shortcuts....you are the MAN!  Just donated $10 to support the cause as well.  Keep it up hermano.

**Reply by Dooti-2** [reply]
- okay..... @mriscoc when my bed hits 39 degrees the machine flashes power and restarts....so odd.  Any suggestions?  

- its the 427 board....E3v2 
- w/ CRTOUCH 
- (i plugged in BLTOUCH, still does it.)
- I reverted back to 20220330 release (or there about) and it's doing it too... 

Should I try the February release?  That was the last stable one I was running.

**Reply by mriscoc** [reply]
Thank you for your donation. Please try to do a "Restore defaults" and "Store settings" to clear old EEPROM values. But, seems to be a possible hardware failure.

**Comment by Dooti-2** [comment]
Yeah it is hardware...I have tried everything and ALL adjustments made via
the screen work EXCEPT for when I begin to heat either the hot end or the
bed.  Checked all cables....even replaced thermistors for good measure.
same thing.  think its the board unfortunately.


On Thu, Apr 21, 2022 at 11:56 AM Miguel Risco-Castillo <
***@***.***> wrote:

> Thank you for your donation. Please try to do a "Restore defaults" and
> "Store settings" to clear old EEPROM values. But, seems to be a possible
> hardware failure.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mriscoc/Ender3V2S1/discussions/127#discussioncomment-2610309>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASLSFDASNU33PDZWL3RCH33VGGCENANCNFSM5T24AMIA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Matt Flier
615.545.3655

**Reply by mriscoc** [reply]
Maybe it is a power source failure, when the bed starts to heat, it consumes more energy.

**Comment by RubeNL1992** [comment]
At first it takes a while to start printing, because preheating will fluctuate to much around the desired temperature.
As soon as its starts printing it looks like its stops heating and cant print anymore.

**Reply by mriscoc** [reply]
There is not that behavior in our tests, check your termistor. Do a PID, also, check your silicone jacket of the heater block    

![image](https://user-images.githubusercontent.com/2745567/165196364-69ba5d05-e149-4ea8-ad9c-2e553eab8930.png)

**Reply by RubeNL1992** [reply]
Termistor fits well! Did a PID aswell but without succes. Just went back to version 20220330 and the problem is solved again.

**Comment by azsundevil02** [comment]
Having a new error on my s1 F4 board. Finished a 3 day print. Went to do a mesh (9x9) and it just keeps looping through the auto bed level process. When it finishes it just re homes and starts again.  Doesn't give me the results screen.  Then it started saying there amwas a probe error shut down machine. Have done that - but I can't seem to get a new abl to run and can't print.

**Reply by azsundevil02** [reply]
After I shut off the machine and turned it back on all settings had been wiped.  Z offset was set to zero and there was no stored mesh in settings to load.  Not sure if I read that the z offset being wiped was a known issue when just rebooting the machine.

**Reply by azsundevil02** [reply]
Hmmm I still had hs mode on or it turned itself back on.  Will try turning that off and trying again.  Do the settings get wiped every time you turn off the machine?  I thought it was just when you flash the firmware.  Maybe I misunderstood.  Also - when printing - my double lines turn blue if I activate leveling...am I supposed to have red at all or just the blue / white?

**Reply by mriscoc** [reply]
F4 is in experimental stage and has known issues https://github.com/mriscoc/Ender3V2S1/issues/125

**Reply by azsundevil02** [reply]
Awesome - thank you.  I don't know why I didn't see that one when I was scrolling thru the list.

**Comment by vpereiradn** [comment]
Thank you so much for the FW I'm using it now, I hit 220x220x250
and the e-steps 424.9 but it is still off the edge and the size of the bed is out coming out the automatic alignment of the CRtouch..

you have the values for this FW right!!

I really appreciate if you send me!!

My Printer Ender3 V2 - 4.2.2

with the extruder spirte extruder Pro kit

**Reply by mriscoc** [reply]
> Thank you so much for the FW I'm using it now, I hit 220x220x250 and the e-steps 424.9 but it is still off the edge and the size of the bed is out coming out the automatic alignment of the CRtouch..
> 
> you have the values for this FW right!!
> 
> I really appreciate if you send me!!
> 
> My Printer Ender3 V2 - 4.2.2
> 
> with the extruder spirte extruder Pro kit

Sorry, I can't understand your problem, if you changed your extruder/fan duct then you must change the physical dimensions / offsets according to your specific setup. I don't have those parameters, maybe other users with the same changes could help you. I recommend posting that requirement in our Facebook for V2: https://www.facebook.com/groups/ender3v2firmware
Please check also these first steps: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Reply by killer-vp** [reply]
então to lutando com isso a mais de uma semana nao sei mais o que fazer!!

**Comment by rombourg** [comment]
Miguel.  Thanks for the good work that you shared.  I have the S1 .  Could you provide the source code specific for the S1.
Thanks Ron

**Reply by azsundevil02** [reply]
https://github.com/mriscoc/Ender3V2S1/releases/tag/20220419

Scroll down to the bottom and look for the version for S1

**Reply by rombourg** [reply]
Thanks - I updated my S1 with the Sprite Extruder Pro.  Updated firmware by changing temp sensor to # 13 and Max temp to 300.  Compiled and flashed .  Appears to be working.

**Reply by mriscoc** [reply]
https://github.com/mriscoc/Special_Configurations here are configuration files for the type 13 thermistor.
and some binaries here: https://github.com/mriscoc/Special_Configurations/releases

**Comment by EXCISIONMODZ** [comment]
ender 3 v2 skr mini e3 v3 y axis wont home

**Reply by mriscoc** [reply]
Check your wiring 
For referece: https://github.com/mriscoc/Ender3V2S1/discussions/131

**Reply by EXCISIONMODZ** [reply]
its correct 
it worked on your last version

**Comment by CaptJocco** [comment]
Hello, is there a way to configure this to work with the Ender 3 Pro and it's standard LCD screen. I have the 4.2.7 board, BLTouch and Sprite Extruder Pro, and this firmware seems to have all of the features I'm looking for, except that I have the Ender 3 Pro. I'm still relatively new to 3D printing, I'm learning quickly but have a long way to go. I just tried compiling my own firmware but I guess I still have more to learn because I got a bunch of errors when compiling and I don't know how to fix them. Any help would be greatly appreciated. Thanks

**Reply by mriscoc** [reply]
Hi, the current requirements are published in the Wiki. There is no support for monochrome LCD screens.

**Reply by CaptJocco** [reply]
Thank you for the reply. I just purchased the upgraded color touch screen (it arrives later today from Amazon). Once I install that, your firmware should work for me at that point, correct? I really hope so, lol, as I'm starting to get to my wits end with figuring this out myself. 

Thanks again
